components/scream/src/share/tests/utils_tests.cpp

```diff
@@ -38,3 +39,34 @@ TEST_CASE("field_layout") {
   REQUIRE ( (superset2==tgt || superset2==tgt_rev) );
   REQUIRE ( (superset3==tgt || superset3==tgt_rev) );
 }
+
+TEST_CASE ("time_stamp") {
+  using namespace scream;
+  using TS = util::TimeStamp;
+
+  TS ts1 (2021,10,12,17,8,30);
+  REQUIRE (ts1.get_years()==2021);
+  REQUIRE (ts1.get_months()==10);
+  REQUIRE (ts1.get_days()==12);
+  REQUIRE (ts1.get_hours()==17);
+  REQUIRE (ts1.get_minutes()==8);
+  REQUIRE (ts1.get_seconds()==30);
+
+  // Julian day = day_of_year.fraction_of_day, with day_of_year=0 at Jan 1st.
+  REQUIRE (ts1.get_julian_day()==(284 + (17*3600+8*60+30)/86400.0));
+
+  REQUIRE (ts1.get_date_string()=="2021-10-12");
+  REQUIRE (ts1.get_time_string()=="17:08:30");
+  REQUIRE (ts1.to_string()=="2021-10-12 17:08:30");
+
+  // Comparisons
```

Comment:
Could you also make a few tests which check that adding 31 days, for example, switches the month.  Or adding 365 days switches the year?  Would that be helpful?

Reply:
Yeah, that's actually useful. I'll add it.